src/MassTransit.ApplicationInsights/ApplicationInsightsSendFilter.cs

```diff
@@ -13,30 +13,41 @@
 namespace MassTransit.ApplicationInsights
 {
 	using System;
+	using System.Linq;
 	using System.Threading.Tasks;
 	using GreenPipes;
 	using Microsoft.ApplicationInsights;
 	using Microsoft.ApplicationInsights.DataContracts;
+	using Microsoft.ApplicationInsights.Extensibility;
 
 	public class ApplicationInsightsSendFilter<T> 
 		: IFilter<T>
 		where T : class, SendContext
 	{
+		const string MessageId = nameof(MessageId);
+		const string ConversationId = nameof(ConversationId);
+		const string CorrelationId = nameof(CorrelationId);

```

Comment:
If `InitiatorId` is present, it's the `CorrelationId` of the message that was being consumed when the current message was published/sent. The whole point is to tie the chain together, ConversationId is the same for all of them, and Correlation/Initator form a linked list.

Reply:
I'm not sure how App Insights glues it all together, I'd need to know that before I could comment on where those values (Corr/Init) should be set.